packages/api/lambdas/kinesis-consumer.js

```diff
@@ -34,63 +36,74 @@ async function getKinesisRules(event) {
+ * @returns {Promise} promise resolved when the message is queued
  */
-async function createOneTimeRules(kinesisRules) {
-  const oneTimeRulePromises = kinesisRules.map((kinesisRule) => {
-    const oneTimeRuleParams = Object.assign({}, kinesisRule);
-    delete oneTimeRuleParams['createdAt'];
-    delete oneTimeRuleParams['updatedAt'];
-    oneTimeRuleParams.name = `${kinesisRule.name}_${Date.now().toString()}`;
-    oneTimeRuleParams.rule.type = 'onetime';
-    return model.create(oneTimeRuleParams);
-  });
+async function queueMessageForRule(kinesisRule, eventObject) {
+  const item = {
+    workflow: kinesisRule.workflow,
+    provider: kinesisRule.provider,
+    collection: kinesisRule.collection,
+    payload: eventObject
+  };
 
-  return await Promise.all(oneTimeRulePromises);
+  return Rule.buildPayload(item)
+    .then((payload) => queueWorkflowMessage(payload));
 }
 
 /**
- * `validateMessage` validates an event as being valid for creating a workflow. See the messageSchema defined at
- * the top of this file.
+ * `validateMessage` validates an event as being valid for creating a workflow.
+ * See the messageSchema defined at the top of this file.
  *
- * @param {object} event lambda event
- * @returns {(error|object)} Throws an Ajv.ValidationError if event object is invalid. Returns the event object if event is valid.
+ * @param {Object} event - lambda event
+ * @returns {(error|Object)} Throws an Ajv.ValidationError if event object is invalid.
+ * Returns the event object if event is valid.
  */
 async function validateMessage(event) {
-  const ajv = new Ajv({allErrors: true});
+  const ajv = new Ajv({ allErrors: true });
   const validate = ajv.compile(messageSchema);
   return await validate(event);
 }
 
+/**
+ * Process data sent to a kinesis stream. Validate the data and
+ * create rules.
+ *
+ * @param {*} record - input to the kinesis stream
+ * @returns {[Promises]} Array of promises. Each promise is resolved when a
+ * message is queued for all associated kinesis rules.
+ */
 async function processRecord(record) {
   const dataBlob = record.kinesis.data;
   const dataString = Buffer.from(dataBlob, 'base64').toString();
   const eventObject = JSON.parse(dataString);
 
-  await validateMessage(eventObject)
+  return validateMessage(eventObject)
     .then(getKinesisRules)
-    .then((kinesisRules) => {
-      return createOneTimeRules(kinesisRules);
-    });
+    .then((kinesisRules) => (
+      Promise.all(kinesisRules.map((kinesisRule) => queueMessageForRule(kinesisRule, eventObject)))
+    ));
 }
 
 /**
- * `handler` Looks up enabled 'kinesis'-type rules associated with the collection in the event argument. It
- * creates new onetime rules for each rule found to trigger the workflow defined in the 'kinesis'-type rule.
+ * `handler` Looks up enabled 'kinesis'-type rules associated with the collection
+ * in the event argument. It creates new onetime rules for each rule found to trigger
```

Comment:
This comment is out of date - `It creates new onetime rules for each rule` -> `It enqueues a message for each kinesis-type rule to trigger the associated workflow.`